pkg/util/containers/v2/metrics/docker/collector.go

```diff
@@ -80,6 +92,59 @@ func (d *dockerCollector) GetContainerNetworkStats(containerID string, cacheVali
 	return convertNetworkStats(stats.Networks), nil
 }
 
+// GetContainerIDForPID returns the container ID for given PID
+func (d *dockerCollector) GetContainerIDForPID(pid int, cacheValidity time.Duration) (string, error) {
+	currentTime := time.Now()
+	strPid := strconv.Itoa(pid)
+
+	cID, found, _ := d.pidCache.Get(currentTime, strPid, cacheValidity)
+	if found {
+		return cID.(string), nil
+	}
+
+	// If we've done a full refresh within cacheValidity, we do not trigger another full refresh
+	_, found, _ = d.pidCache.Get(currentTime, pidCacheFullRefreshKey, cacheValidity)
+	if found {
+		return "", nil
+	}
+
+	// Full refresh
+	containers, err := d.metadataStore.ListContainers()
+	if err != nil {
+		return "", nil
```

Comment:
This swallows errors as well, I think we should either return the `err` or log it.

Reply:
I agree